balancer-js/src/modules/pools/poolFactory.module.integration.spec.ts

```diff
@@ -0,0 +1,160 @@
+  symbol: 'WPOOL',
+  initialFee: '0.1',
+  seedTokens: SEED_TOKENS,
+  owner: '0x0000000000000000000000000000000000000001',
+  value: '0.1',
+};
+const poolFactoryContract = '';
+const getERC20Contract = (address: string) => {
+  return new ethers.Contract(
+    address,
+    [
+      'function balanceOf(address) view returns (uint256)',
+      'function name() returns (string)',
+    ],
+    provider
+  );
+};
+
+describe('pool factory module', () => {
+  // Setup chain
+  before(async function () {
+    this.timeout(20000);
+
+    await provider.send('hardhat_reset', [
+      {
+        forking: {
+          jsonRpcUrl,
+        },
+      },
+    ]);
+  });
+
+  context('factory transaction', () => {
+    let balancer: BalancerSDK,
+      to: string,
+      data: string,
+      value: BigNumber,
+      expectedPoolName: string,
+      transactionReceipt: ethers.providers.TransactionReceipt;
+    beforeEach(async () => {
+      balancer = new BalancerSDK(sdkConfig);
+      const createTx = await balancer.pools.weighted.buildCreateTx(POOL_PARAMS);
+      to = createTx.to;
+      data = createTx.data;
+      value = createTx.value as BigNumber;
+      expectedPoolName = createTx.attributes.name;
+      const tx = { to, data, value };
+      transactionReceipt = await (await signer.sendTransaction(tx)).wait();
+    });
+
+    it('should send the transaction succesfully', async () => {
+      expect(transactionReceipt.status).to.eql(1);
+    });
+
+    it('should create a pool with the correct token name', async () => {
+      const abi = ['event PoolCreated(address indexed)'];
+      const contract = new Contract(poolFactoryContract, abi, provider);
+      const { address } = contract.filters.PoolCreated();
+      const tokenContract = getERC20Contract(address as string);
+      expect(expectedPoolName).to.eql(await tokenContract.getName());
+    });
+  });
+  context('initial join transaction', () => {
+    let balancer: BalancerSDK,
+      transactionReceipt: ethers.providers.TransactionReceipt;
+    beforeEach(async () => {
+      balancer = new BalancerSDK(sdkConfig);
+      const txAttributes = await balancer.pools.weighted.buildCreateTx(
+        POOL_PARAMS
+      );
+      transactionReceipt = await (
+        await signer.sendTransaction(txAttributes)
+      ).wait();
+    });
+
+    it('should send the transaction succesfully', async () => {
+      expect(transactionReceipt.status).to.eql(1);
+    });
+
+    it('should give user tokens on initial join', async () => {
+      const { address } = await balancer.pools.getPoolInfoFromCreateTx(
+        transactionReceipt
+      );
+      const tx = await balancer.pools.weighted.buildInitJoin(INIT_JOIN_PARAMS);
+      await (await signer.sendTransaction(tx)).wait();
+      const createdPool = getERC20Contract(address);
+      const senderBalance: BigNumber = await createdPool.balanceOf(signer);
+      expect(senderBalance.gt(BigNumber.from('0'))).to.eql(true);
+    });
+  });
```

Comment:
Can we also test the tokensIn have only decreased by expected amount.

Reply:
Added a new test for that